Got Error: Cannot find module './task-names' when using 0.3.3

```
npx hardhat compile
An unexpected error occurred:

Error: Cannot find module './task-names'
Require stack:
- /Users/jishankai/PoLeVeDi/SignalPlus/node_modules/@matterlabs/hardhat-zksync-deploy/dist/index.js
- /Users/jishankai/PoLeVeDi/SignalPlus/hardhat.config.js
- /Users/jishankai/PoLeVeDi/SignalPlus/node_modules/hardhat/internal/core/config/config-loading.js
- /Users/jishankai/PoLeVeDi/SignalPlus/node_modules/hardhat/internal/cli/cli.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:815:15)
    at Function.Module._load (internal/modules/cjs/loader.js:667:27)
    at Module.require (internal/modules/cjs/loader.js:887:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at Object.<anonymous> (/Users/jishankai/PoLeVeDi/SignalPlus/node_modules/@matterlabs/hardhat-zksync-deploy/dist/index.js:18:22)
    at Module._compile (internal/modules/cjs/loader.js:999:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)
    at Module.load (internal/modules/cjs/loader.js:863:32)
    at Function.Module._load (internal/modules/cjs/loader.js:708:14)
    at Module.require (internal/modules/cjs/loader.js:887:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/Users/jishankai/PoLeVeDi/SignalPlus/node_modules/@matterlabs/hardhat-zksync-deploy/dist/index.js',
    '/Users/jishankai/PoLeVeDi/SignalPlus/hardhat.config.js',
    '/Users/jishankai/PoLeVeDi/SignalPlus/node_modules/hardhat/internal/core/config/config-loading.js',
    '/Users/jishankai/PoLeVeDi/SignalPlus/node_modules/hardhat/internal/cli/cli.js'
  ]
}
```

Fixed if downgrading it to 0.1.

--- a/examples/basic-example/hardhat.config.ts
+++ b/examples/basic-example/hardhat.config.ts
@@ -1,14 +1,16 @@
-require("@matterlabs/hardhat-zksync-deploy");
-require("@matterlabs/hardhat-zksync-solc");
+import "@matterlabs/hardhat-zksync-deploy";
+import "@matterlabs/hardhat-zksync-solc";
+import { HardhatUserConfig } from 'hardhat/config';
 
-module.exports = {
+const config: HardhatUserConfig = {
   zksolc: {
     version: "0.1.0",
-    compilerSource: "binary",
+    compilerSource: "docker",
     settings: {
-      optimizer: {
-        enabled: true,
-      },
+      experimental: {
+        dockerImage: "matterlabs/zksolc",
+        tag: "v1.1.0"
+      }
     },
   },
   zkSyncDeploy: {
@@ -24,3 +26,5 @@ module.exports = {
     version: "0.8.12"
   }
 };
+
+export default config;

--- a/packages/hardhat-zksync-deploy/package.json
+++ b/packages/hardhat-zksync-deploy/package.json
@@ -1,6 +1,6 @@
 {
   "name": "@matterlabs/hardhat-zksync-deploy",
-  "version": "0.3.3",
+  "version": "0.3.5",
   "description": "Hardhat plugin to deploy smart contracts into the zkSync network",
   "repository": "github:matter-labs/hardhat-zksync",
   "homepage": "https://github.com/matter-labs/hardhat-zksync/tree/main/packages/hardhat-zksync-deploy",
@@ -26,7 +26,7 @@
     "clean": "rimraf dist"
   },
   "files": [
-    "dist/src/",
+    "dist/",
     "src/",
     "LICENSE",
     "README.md"

--- a/packages/hardhat-zksync-solc/package.json
+++ b/packages/hardhat-zksync-solc/package.json
@@ -1,6 +1,6 @@
 {
   "name": "@matterlabs/hardhat-zksync-solc",
-  "version": "0.3.5",
+  "version": "0.3.6",
   "description": "Hardhat plugin to compile smart contracts for the zkSync network",
   "repository": "github:matter-labs/hardhat-zksync",
   "homepage": "https://github.com/matter-labs/hardhat-zksync/tree/main/packages/hardhat-zksync-solc",

--- a/packages/hardhat-zksync-solc/src/compile/docker.ts
+++ b/packages/hardhat-zksync-solc/src/compile/docker.ts
@@ -126,8 +126,7 @@ export async function compileWithDocker(
     input: CompilerInput,
     docker: HardhatDocker,
     image: Image,
-    config: ZkSolcConfig
-): Promise<any> {
+) {
     const command = ['zksolc', '--standard-json'];
 
     // @ts-ignore

--- a/packages/hardhat-zksync-solc/src/compile/index.ts
+++ b/packages/hardhat-zksync-solc/src/compile/index.ts
@@ -12,10 +12,6 @@ import { CompilerInput } from 'hardhat/types';
 import { pluginError } from '../utils';
 
 export async function compile(zksolcConfig: ZkSolcConfig, input: CompilerInput, solcPath?: string) {
-    if (zksolcConfig.settings.optimizer) {
-        console.warn('Settings for optimizer are deprecated, optimizer is enabled in any case.');
-    }
-
     let compiler: ICompiler;
     if (zksolcConfig.compilerSource == 'binary') {
         if (solcPath == null) {
@@ -38,7 +34,7 @@ export interface ICompiler {
 export class BinaryCompiler implements ICompiler {
     constructor(public solcPath: string) {}
 
-    public async compile(input: CompilerInput, config: ZkSolcConfig): Promise<any> {
+    public async compile(input: CompilerInput, config: ZkSolcConfig) {
         return await compileWithBinary(input, config, this.solcPath);
     }
 }
@@ -56,7 +52,7 @@ export class DockerCompiler implements ICompiler {
         return new DockerCompiler(image, docker);
     }
 
-    public async compile(input: CompilerInput, config: ZkSolcConfig): Promise<any> {
-        return await compileWithDocker(input, this.docker, this.dockerImage, config);
+    public async compile(input: CompilerInput, _config: ZkSolcConfig) {
+        return await compileWithDocker(input, this.docker, this.dockerImage);
     }
 }

--- a/packages/hardhat-zksync-solc/src/index.ts
+++ b/packages/hardhat-zksync-solc/src/index.ts
@@ -5,7 +5,7 @@ import {
 } from 'hardhat/builtin-tasks/task-names';
 import { extendEnvironment, extendConfig, subtask } from 'hardhat/internal/core/config/config-env';
 import './type-extensions';
-import { FactoryDeps } from './types';
+import { FactoryDeps, ZkSolcConfig } from './types';
 import { Artifacts, getArtifactFromContractOutput } from 'hardhat/internal/artifacts';
 import { compile } from './compile';
 import { zeroxlify, pluginError } from './utils';
@@ -14,21 +14,22 @@ import semver from 'semver';
 
 const ZK_ARTIFACT_FORMAT_VERSION = 'hh-zksolc-artifact-1';
 
-extendConfig((config) => {
-    const defaultConfig = {
+extendConfig((config, userConfig) => {
+    const defaultConfig: ZkSolcConfig = {
         version: 'latest',
         compilerSource: 'binary',
         settings: {
             compilerPath: 'zksolc',
-            optimizer: {
-                enabled: false,
-            },
             experimental: {},
         },
     };
 
-    config.zksolc = { ...defaultConfig, ...config.zksolc };
-    config.zksolc.settings = { ...defaultConfig.settings, ...config.zksolc.settings };
+    if (userConfig?.zksolc?.settings?.optimizer) {
+        console.warn('`optimizer` setting is deprecated, optimizer is always enabled');
+    }
+
+    config.zksolc = { ...defaultConfig, ...userConfig?.zksolc };
+    config.zksolc.settings = { ...defaultConfig.settings, ...userConfig?.zksolc?.settings };
 });
 
 extendEnvironment((hre) => {
@@ -51,7 +52,7 @@ extendEnvironment((hre) => {
         (hre as any).artifacts = new Artifacts(artifactsPath);
 
         // If solidity optimizer is not enabled, the libraries are not inlined and
-        // we have to manually pass them into zksolc. So for now we force the optimization.
+        // we have to manually pass them into zksolc. That's why we force the optimization.
         hre.config.solidity.compilers.forEach((compiler) => {
             let settings = compiler.settings || {};
             compiler.settings = { ...settings, optimizer: { enabled: true } };
@@ -140,7 +141,7 @@ subtask(TASK_COMPILE_SOLIDITY_GET_SOLC_BUILD, async (args: { solcVersion: string
     const solcBuild = await runSuper(args);
     const compilerPath = hre.config.zksolc.settings.compilerPath;
 
-    const versionOutput = spawnSync(compilerPath, ['--version']);
+    const versionOutput = spawnSync(compilerPath!, ['--version']);
     const version = versionOutput.stdout?.toString().match(/\d+\.\d+\.\d+/)?.toString();
 
     if (versionOutput.status !== 0 || version == null) {

--- a/packages/hardhat-zksync-solc/src/types.ts
+++ b/packages/hardhat-zksync-solc/src/types.ts
@@ -6,7 +6,7 @@ export interface ZkSolcConfig {
     settings: {
         // Path to zksolc binary. If compilerSource == "docker", this option is ignored.
         // By default, zksolc in $PATH is used.
-        compilerPath: string;
+        compilerPath?: string;
         /**
          * @deprecated This property has no effect, optimizer is always enabled. It has been maintained for compatibility purpose.
          */

--- a/packages/hardhat-zksync-solc/test/fixture-projects/factory/hardhat.config.js
+++ b/packages/hardhat-zksync-solc/test/fixture-projects/factory/hardhat.config.js
@@ -5,9 +5,6 @@ module.exports = {
     version: "0.1.0",
     compilerSource: "docker",
     settings: {
-      optimizer: {
-        enabled: true,
-      },
       experimental: {
         dockerImage: "matterlabs/zksolc"
       }

--- a/packages/hardhat-zksync-solc/test/fixture-projects/library/hardhat.config.js
+++ b/packages/hardhat-zksync-solc/test/fixture-projects/library/hardhat.config.js
@@ -5,9 +5,6 @@ module.exports = {
     version: "0.1.0",
     compilerSource: "docker",
     settings: {
-      optimizer: {
-        enabled: true,
-      },
       experimental: {
         dockerImage: "matterlabs/zksolc"
       }

--- a/packages/hardhat-zksync-solc/test/fixture-projects/linked/hardhat.config.js
+++ b/packages/hardhat-zksync-solc/test/fixture-projects/linked/hardhat.config.js
@@ -5,9 +5,6 @@ module.exports = {
     version: "0.1.0",
     compilerSource: "docker",
     settings: {
-      optimizer: {
-        enabled: true,
-      },
       experimental: {
         dockerImage: "matterlabs/zksolc"
       },

--- a/packages/hardhat-zksync-solc/test/fixture-projects/nested/hardhat.config.js
+++ b/packages/hardhat-zksync-solc/test/fixture-projects/nested/hardhat.config.js
@@ -5,9 +5,6 @@ module.exports = {
     version: "0.1.0",
     compilerSource: "docker",
     settings: {
-      optimizer: {
-        enabled: true,
-      },
       experimental: {
         dockerImage: "matterlabs/zksolc"
       }

--- a/packages/hardhat-zksync-solc/test/fixture-projects/simple/hardhat.config.js
+++ b/packages/hardhat-zksync-solc/test/fixture-projects/simple/hardhat.config.js
@@ -5,9 +5,6 @@ module.exports = {
     version: "0.1.0",
     compilerSource: "docker",
     settings: {
-      optimizer: {
-        enabled: true,
-      },
       experimental: {
         dockerImage: "matterlabs/zksolc"
       }

